src/CTA.WebForms2Blazor/ControlConverters/ControlConverter.cs

```diff
@@ -9,7 +9,38 @@ namespace CTA.WebForms2Blazor.ControlConverters
 {
     public abstract class ControlConverter
     {
+        protected class Attribute
+        {
+            public string Name { get; }
+            public string Value { get; }
+
+            public Attribute(string name, string value)
+            {
+                Name = name;
+                Value = value;
+            }
+
+            public override string ToString()
+            {
+                return Name + "=" + Value;
+            }
+
+            public bool Equals(Attribute other)
+            {
+                if (other is null)
+                {
+                    return false;
+                }
+                return this.Name == other.Name;
+            }
+            public override bool Equals(object obj) => Equals(obj as Attribute);
+            public override int GetHashCode() => (Name).GetHashCode();
+        }
         protected abstract Dictionary<String, String> AttributeMap { get; }
+        protected virtual IEnumerable<Attribute> NewAttributes
```

Comment:
Any reason we want to always return null for NewAttributes?

Reply:
Not really, its just that the function it gets passed to can handle a null NewAttributes value but I've changed it to avoid future issues